pandas/core/internals.py

```diff
@@ -109,14 +111,15 @@ class Block(PandasObject):
     def __init__(self, values, placement, ndim=None):
         if ndim is None:
             ndim = values.ndim
-        elif values.ndim != ndim:
+        elif self._validate_ndim and values.ndim != ndim:
```

Comment:
make a method ``_validate_ndim(ndim)`` to do this

Reply:
Hmm, I'd find it surprising to call a method like `self._validate_ndim` and have it not actually do validation. We could call it `_maybe_validate_ndim`, but then you have to know why it's 'maybe'...

Reply:
my point is its much better to have these as methods and not properties

Reply:
I'm probably misunderstanding. https://github.com/pandas-dev/pandas/pull/19268/commits/9c06b13d2c8ecf53d678a0ec613acea927bb3955 moved that check to a method, but I don't think it's clearer.

Note that `NonConsolidatableMixin` already had the `_validate_ndim` property. The only reason we need to add the `if self._validate_ndim` is since all child blocks now eventually call `Block.__init__`. The whole `_validate_ndim` stuff seems like a code smell to be anyway, but that cleanup needn't be done in this PR I think.